src/Functions/LowerUpperUTF8Impl.h

```diff
@@ -24,6 +24,24 @@ extern const int LOGICAL_ERROR;
 template <char not_case_lower_bound, char not_case_upper_bound, bool upper>
 struct LowerUpperUTF8Impl
 {
+    struct CaseMapHolder
```

Comment:
I feel it'd be simpler to use `SCOPE_EXIT` or std::unique_ptr with a custom delete function.

Reply:
Agree, fixed.